src/lib.rs

```diff
@@ -64,10 +77,26 @@ pub fn custom_rocket(
         .manage(auth_service)
         .manage(notifier)
         .manage(jwt_service)
-        .manage(ServiceSecretsStore::default())
+        .manage(service_secrets_store)
+}
+
+async fn sequential_fairings(rocket: &Rocket<Build>) -> Result<(), String> {
+    run_migrations(rocket).await?;
+    // Wait for the XMPP server to finish starting up (1 second should be more than enough)
+    sleep(Duration::from_secs(1)).await;
```

Comment:
Startup is longer by one second, but at least its consistent. If we really want to we could find a way for the server to send a signal to the API, but we won't bother for now. It's not like this will grow over time and bite us later.